sklearn/neighbors/classification.py

```diff
@@ -375,8 +382,57 @@ def __init__(self, radius=1.0, weights='uniform',
         self.weights = _check_weights(weights)
         self.outlier_label = outlier_label
 
+    def fit(self, X, y):
+        """Fit the model using X as training data and y as target values
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix, BallTree, KDTree}
+            Training data. If array or matrix, shape [n_samples, n_features],
+            or [n_samples, n_samples] if metric='precomputed'.
+
+        y : {array-like, sparse matrix}
+            Target values of shape = [n_samples] or [n_samples, n_outputs]
+
+        """
+
+        SupervisedIntegerMixin.fit(self, X, y)
+
+        classes_ = self.classes_
+        _y = self._y
+        if not self.outputs_2d_:
+            _y = self._y.reshape((-1, 1))
+            classes_ = [self.classes_]
+
+        if self.outlier_label is None:
+            outlier_label_ = None
+        elif self.outlier_label == 'most_frequent':
+            outlier_label_ = []
+            for k, classes_k in enumerate(classes_):
```

Comment:
Maybe add a small comment here to clarify that it does not iterate over classes, but over outputs.

Reply:
Added